notifications_utils/logging.py

```diff
@@ -17,11 +19,63 @@
 logger = logging.getLogger(__name__)
 
 
+def _common_request_extra_log_context():
+    return {
+        "method": request.method,
+        "url": request.url,
+        "endpoint": request.endpoint,
+        # pid and is available on LogRecord by default, as `process` but I don't see
+        # a straightforward way of selectively including it only in certain log messages -
+        # it is designed to be included when the formatter is being configured. This is
+        # why I'm manually grabbing it and putting it in as `extra` here, avoiding the
+        # existing parameter name to prevent LogRecord from complaining
+        "process_": getpid(),
+    }
+
+
 def init_app(app, statsd_client=None, extra_filters: Sequence[logging.Filter] = tuple()):
     app.config.setdefault("NOTIFY_LOG_LEVEL", "INFO")
     app.config.setdefault("NOTIFY_APP_NAME", "none")
     app.config.setdefault("NOTIFY_LOG_PATH", "./log/application.log")
     app.config.setdefault("NOTIFY_RUNTIME_PLATFORM", None)
+    app.config.setdefault(
+        "NOTIFY_REQUEST_LOG_LEVEL",
+        "WARNING" if app.config["NOTIFY_RUNTIME_PLATFORM"] == "paas" else "NOTSET",
```

Comment:
I just want to check I have understood things correctly here. With the above defaults, when we release this code to our apps we expect to see the following changes

- On the PaaS, if an app serves a 5xx response then it will also issue a warning level log line corresponding
- On the PaaS, if an app services a 2xx response then it will not issue a log line (because log level is info and is less than warning)
- When running locally, if an app serves any response then it will log the the 'before request' log and the 'after request' log
- When runnning in ECS, if an app serves any response then it will log the the 'before request' log and the 'after request' log

If so, it seems reasonable that for ECS by default we will want lots of logging. It also seems reasonable for local development too.

I would be open to not having PaaS giving us extra warning logs for 5xx responses (we already get these from the gorouter) but happy to do so if you think it is useful.

Reply:
> When running locally, if an app serves any response then it will log the the 'before request' log and the 'after request' log

> When runnning in ECS, if an app serves any response then it will log the the 'before request' log and the 'after request' log

Both only if someone has set their `NOTIFY_LOG_LEVEL` to `DEBUG`

> I would be open to not having PaaS giving us extra warning logs for 5xx responses (we already get these from the gorouter) but happy to do so if you think it is useful.

Meh I suppose I could default `NOTIFY_REQUEST_LOG_LEVEL` to `CRITICAL` if we're on paas.

Reply:
Ah yes, I hadn't clocked the fact that this also depends on `NOTIFY_LOG_LEVEL` as the parent logger. Thanks, the above makes sense and helps clarify my understanding.